# **Exploratory Data Analysis (EDA) Version 1**


In [1]:
import pandas as pd
import numpy as np

# Initial Analysis #

In [2]:
dt_frame = pd.read_csv("../input/top-1000-highest-grossing-movies/Highest Holywood Grossing Movies.csv")

print(str.format("Number of data points: {0}, number of features: {1}",dt_frame.shape[0], dt_frame.shape[1]))
print('\n'+'-'*50+'\n')
print(str.format("List of features: {0}",dt_frame.columns))

dt_frame.head(3)

Number of data points: 918, number of features: 11

--------------------------------------------------

List of features: Index(['Unnamed: 0', 'Title', 'Movie Info', 'Distributor', 'Release Date',
       'Domestic Sales (in $)', 'International Sales (in $)',
       'World Sales (in $)', 'Genre', 'Movie Runtime', 'License'],
      dtype='object')


Unnamed: 0                                              Title  \
0           0  Star Wars: Episode VII - The Force Awakens (2015)   
1           1                           Avengers: Endgame (2019)   
2           2                                      Avatar (2009)   

                                          Movie Info  \
0  As a new threat to the galaxy rises, Rey, a de...   
1  After the devastating events of Avengers: Infi...   
2  A paraplegic Marine dispatched to the moon Pan...   

                           Distributor       Release Date  \
0  Walt Disney Studios Motion Pictures  December 16, 2015   
1  Walt Disney Studios Motion Pictures     April 24, 2019   
2                Twentieth Century Fox  December 16, 2009   

   Domestic Sales (in $)  International Sales (in $)  World Sales (in $)  \
0              936662225                  1132859475          2069521700   
1              858373000                  1939128328          2797501328   
2              760507625                  2086738578          2847246203   

                                          Genre Movie Runtime License  
0             ['Action', 'Adventure', 'Sci-Fi']   2 hr 18 min   PG-13  
1    ['Action', 'Adventure', 'Drama', 'Sci-Fi']    3 hr 1 min   PG-13  
2  ['Action', 'Adventure', 'Fantasy', 'Sci-Fi']   2 hr 42 min   PG-13

In [3]:
print(str.format("Records where the World sales does not matches with the sales of Domestic and International sales: {0}", 
                (dt_frame[dt_frame['World Sales (in $)'] != (dt_frame['Domestic Sales (in $)'] + dt_frame['International Sales (in $)'])].shape[0])))

Records where the World sales does not matches with the sales of Domestic and International sales: 29


In [4]:
dt_frame.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 918 entries, 0 to 917
Data columns (total 11 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   Unnamed: 0                  918 non-null    int64 
 1   Title                       918 non-null    object
 2   Movie Info                  918 non-null    object
 3   Distributor                 918 non-null    object
 4   Release Date                800 non-null    object
 5   Domestic Sales (in $)       918 non-null    int64 
 6   International Sales (in $)  918 non-null    int64 
 7   World Sales (in $)          918 non-null    int64 
 8   Genre                       918 non-null    object
 9   Movie Runtime               918 non-null    object
 10  License                     744 non-null    object
dtypes: int64(4), object(7)
memory usage: 79.0+ KB


In [5]:
dt_frame[dt_frame.isna().any(axis=1)]

Unnamed: 0                             Title  \
3             3              Black Panther (2018)   
4             4     Avengers: Infinity War (2018)   
5             5    Spider-Man: No Way Home (2021)   
10           10              Incredibles 2 (2018)   
16           16               Finding Dory (2016)   
..          ...                               ...   
891         891         Mr. Holland's Opus (1995)   
896         896       Bridge to Terabithia (2007)   
903         903                       MASH (1970)   
912         912                      Congo (1995)   
914         914  Jimmy Neutron: Boy Genius (2001)   

                                            Movie Info  \
3    T'Challa, heir to the hidden but advanced king...   
4    The Avengers and their allies must be willing ...   
5    With Spider-Man's identity now revealed, Peter...   
10   The Incredibles family takes on a new mission ...   
16   Friendly but forgetful blue tang Dory begins a...   
..                                                 ...   
891  A frustrated composer finds fulfillment as a h...   
896  A preteen's life turns upside down when he bef...   
903  The staff of a Korean War field hospital use h...   
912  When an expedition to the African Congo ends i...   
914  An eight-year-old boy genius and his friends m...   

                             Distributor       Release Date  \
3    Walt Disney Studios Motion Pictures                NaN   
4    Walt Disney Studios Motion Pictures                NaN   
5      Sony Pictures Entertainment (SPE)                NaN   
10   Walt Disney Studios Motion Pictures                NaN   
16   Walt Disney Studios Motion Pictures                NaN   
..                                   ...                ...   
891  Walt Disney Studios Motion Pictures                NaN   
896  Walt Disney Studios Motion Pictures                NaN   
903                Twentieth Century Fox                NaN   
912                   Paramount Pictures                NaN   
914                   Paramount Pictures  December 21, 2001   

     Domestic Sales (in $)  International Sales (in $)  World Sales (in $)  \
3                700426566                   647171407          1347597973   
4                678815482                  1369544272          2048359754   
5                675813257                   868642706          1544455963   
10               608581744                   634507500          1243089244   
16               486295561                   542275381          1028570942   
..                     ...                         ...                 ...   
891               82569971                    23700000           106269971   
896               82272442                    55314621           137587063   
903               81600000                    81600000            81600000   
912               81022101                    71000000           152022101   
914               80936232                    22056304           102992536   

                                                 Genre Movie Runtime License  
3                    ['Action', 'Adventure', 'Sci-Fi']   2 hr 14 min     NaN  
4                    ['Action', 'Adventure', 'Sci-Fi']   2 hr 29 min     NaN  
5         ['Action', 'Adventure', 'Fantasy', 'Sci-Fi']   2 hr 28 min     NaN  
10   ['Action', 'Adventure', 'Animation', 'Comedy',...   1 hr 58 min     NaN  
16      ['Adventure', 'Animation', 'Comedy', 'Family']   1 hr 37 min     NaN  
..                                                 ...           ...     ...  
891                                 ['Drama', 'Music']   2 hr 23 min     NaN  
896                     ['Drama', 'Family', 'Fantasy']   1 hr 36 min     NaN  
903                         ['Comedy', 'Drama', 'War']   1 hr 56 min     NaN  
912       ['Action', 'Adventure', 'Mystery', 'Sci-Fi']   1 hr 49 min     NaN  
914  ['Action', 'Adventure', 'Animation', 'Comedy',...   1 hr 22 min     NaN  

[174 rows x 11 columns]

In [6]:
#Calculate percenatge of null value in the respective columns.
missing_columns = dt_frame.isnull().sum() / len(dt_frame) * 100
pd.DataFrame({'Missing Values %': missing_columns})

Missing Values %
Unnamed: 0                          0.000000
Title                               0.000000
Movie Info                          0.000000
Distributor                         0.000000
Release Date                       12.854031
Domestic Sales (in $)               0.000000
International Sales (in $)          0.000000
World Sales (in $)                  0.000000
Genre                               0.000000
Movie Runtime                       0.000000
License                            18.954248

In [7]:
# Number of unique movie Distributor
dt_frame.sort_values("Distributor", inplace = True)
dt_frame.Distributor.unique()

array(['20th Century Studios', 'Artisan Entertainment',
       'Columbia Pictures', 'Dimension Films', 'DreamWorks',
       'DreamWorks Distribution', 'FilmDistrict', 'Focus Features',
       'Fox Searchlight Pictures', 'IFC Films', 'Lionsgate',
       'Metro-Goldwyn-Mayer (MGM)', 'Miramax', 'New Line Cinema',
       'Newmarket Films', 'Orion Pictures', 'Paramount Pictures',
       'Relativity Media', 'Revolution Studios', 'Roadside Attractions',
       'STX Entertainment', 'Screen Gems', 'Sony Pictures Classics',
       'Sony Pictures Entertainment (SPE)', 'Summit Entertainment',
       'The Weinstein Company', 'TriStar Pictures',
       'Twentieth Century Fox', 'USA Films', 'United Artists',
       'United Artists Releasing', 'Universal Pictures',
       'Walt Disney Studios Motion Pictures', 'Warner Bros.'],
      dtype=object)

In [8]:
dt_frame[dt_frame['Distributor'] == 'DreamWorks Distribution'].head(2)

Unnamed: 0                       Title  \
179         179  Saving Private Ryan (1998)   
701         701           Collateral (2004)   

                                            Movie Info  \
179  Following the Normandy Landings, a group of U....   
701  A cab driver finds himself the hostage of an e...   

                 Distributor    Release Date  Domestic Sales (in $)  \
179  DreamWorks Distribution   July 24, 1998              217049603   
701  DreamWorks Distribution  August 5, 2004              101005703   

     International Sales (in $)  World Sales (in $)  \
179                   265300000           482349603   
701                   119234222           220239925   

                                        Genre Movie Runtime License  
179                          ['Drama', 'War']   2 hr 49 min       R  
701  ['Action', 'Crime', 'Drama', 'Thriller']          2 hr       R

In [9]:
dt_frame[dt_frame['Distributor'] == 'DreamWorks'].head(2)

Unnamed: 0                   Title  \
72          72  Shrek the Third (2007)   
74          74     Transformers (2007)   

                                           Movie Info Distributor  \
72  Reluctantly designated as the heir to the land...  DreamWorks   
74  An ancient struggle between two Cybertronian r...  DreamWorks   

     Release Date  Domestic Sales (in $)  International Sales (in $)  \
72   May 17, 2007              322719944                   490647436   
74  June 28, 2007              319246193                   390463587   

    World Sales (in $)                                              Genre  \
72           813367380  ['Adventure', 'Animation', 'Comedy', 'Family',...   
74           709709780                  ['Action', 'Adventure', 'Sci-Fi']   

   Movie Runtime License  
72   1 hr 33 min      PG  
74   2 hr 24 min   PG-13

In [10]:
dt_frame.Genre.apply(type).head()

529    <class 'str'>
411    <class 'str'>
408    <class 'str'>
137    <class 'str'>
519    <class 'str'>
Name: Genre, dtype: object

# Observation: #
1. 'Unnamed: 0' feature is an index column and can be droped. 'Movie Info' feature can be droped, this feature tell us brief about the story of the movie and is not much relevent.

2. Renaming features will be lot easier when we need to refer those column name dataframe
    - Rename features to remove spaces. 
    - Remove **Dollar** sign from all sales features. Since all the 'Sales' are in Dollar we can remove '$' from the column name. 
    
3. 'Genre' feature is string representation of a list. So need to convert it in to list and convert them to wide format for easier analysis

4. Seperate 'Release Date' feature as seperate feature as 'Year','Month'.

5. In 'Title' feature year in which the moview was release is added. Since we already have that same detail in the 'Release Date' feature column we can remove those from the 'Title'.

6. 'World Sales' feature is the total sales of the movie. So they should be sum of 'Domestic Sales' and 'International Sales' 
        i.e
            World Sales = Domestic Sales + International Sales    
    Found 29 records where the 'World Sales' does not matches with the sum of 'Domestic' and 'International' sales figure. Need to fix those records.

7. 'Movie Runtime' is in format of hours and minute. Convert them to minute.

8. Same Distributor different naming convention. 
    - 'DreamWorks Distribution' and 'DreamWorks' are same. 
    - 'United Artists' and 'United Artists Releasing' are same.
    - 'Sony Pictures Entertainment (SPE)' is parent company of 'Sony Pictures Classic'. So we can consider them as single company as 'Sony Pictures Entertainment (SPE)'

9. There are 18.8% of records having null value in 'License'. We can either replace those values based on the 'Genre' type of the moview.

10. There are 12.85% records having null in 'Release Date' feature. Since this is an importance feature and we can not simply drop those records. We could get the Year in which moview was release by extracting the year from the 'Title' feature. But we also need the month in which the moview was release, to understand if there is any relationship between the Month a movies are release and there Sales. 
*For these missing values in 'Release Date' feature, have searched and assigned the right release dates for those moviews and created new file '../input/grossing-movies-moviesmissing-release-date-add/Highest_Holywood_Grossing_Movies_v2.csv' that has no **null** values in that column. For futher analysis will be using this file only*

# Data Cleaning
Will clean all points that we have observed in the initial analysis.

In [11]:
dt_frame = pd.read_csv("../input/grossing-movies-moviesmissing-release-date-add/Highest_Holywood_Grossing_Movies_v2.csv")

In [12]:
# Drop 'Unnamed: 0' and 'Movie Info' feature
dt_frame.drop(['Unnamed: 0','Movie Info'], axis=1, inplace = True)

In [13]:
# Rename features, will be lot easier when we need to refer those column name for any analisies.
dt_frame.rename(columns={"Release Date": "Release_Date", 
                         "Domestic Sales (in $)": "Domestic_Sales",
                         "International Sales (in $)": "International_Sales",
                         "World Sales (in $)": "World_Sales",
                         "Movie Runtime": "Duration_min",
                        } 
                , inplace = True )

In [14]:
# Records where the 'World Sales' does not matches with the sum of 'Domestic' and 'International' sales figure
Index = dt_frame[dt_frame['World_Sales'] != (dt_frame['Domestic_Sales'] + dt_frame['International_Sales'])].index
dt_frame.iloc[Index,5] = dt_frame.iloc[Index,3] + dt_frame.iloc[Index,4]

#Verify if the Sales figure is fixed for World_Sales
print(str.format("Number of records having incorrect world sales figure: {0}",dt_frame[dt_frame['World_Sales'] != (dt_frame['Domestic_Sales'] + dt_frame['International_Sales'])].shape[0]))

Number of records having incorrect world sales figure: 0


In [15]:
# 'Genre' feature. Convert it to long format
import ast 
dt_frame['index'] = dt_frame['Genre'].apply(ast.literal_eval)
dt_temp = dt_frame.explode('index', ignore_index = True)
dt_temp['Values'] = 1
genre_series = dt_frame['index']

pivt = dt_temp.pivot_table(index=['Title', 'Distributor', 'Genre','Domestic_Sales', 'International_Sales','World_Sales', 'Duration_min', 'License', 'Release_Date'], 
                           columns='index', 
                           values='Values',
                           fill_value=0)
dt_frame = pivt.reset_index()
dt_frame.index.name = ""
dt_frame.rename(columns={
                         'Action': 'Genre_Action', 'Adventure': 'Genre_Adventure', 'Animation': 'Genre_Animation',
                         'Biography' : 'Genre_Biography',
                         'Comedy' : 'Genre_Comedy', 'Crime' : 'Genre_Crime', 
                         'Documentary' : 'Genre_Documentary', 'Drama': 'Genre_Drama', 
                         'Family': 'Genre_Family', 'Fantasy': 'Genre_Fantasy',
                         'History': 'Genre_History', 'Horror': 'Genre_Horror', 
                         'Music':'Genre_Music', 'Musical':'Genre_Musical', 'Mystery':'Genre_Mystery', 
                         'Romance':'Genre_Romance', 
                         'Sci-Fi':'Genre_Sci-Fi', 'Sport':'Genre_Sport', 
                         'Thriller':'Genre_Thriller', 
                         'War':'Genre_War', 'Western':'Genre_Western'                         
                        } 
                    , inplace = True )
                    
dt_frame.Genre = genre_series
dt_frame.head()

index                    Title                        Distributor  \
                                                                    
0             10,000 BC (2008)                       Warner Bros.   
1                  1917 (2019)                 Universal Pictures   
2      2 Fast 2 Furious (2003)                 Universal Pictures   
3                  2012 (2009)  Sony Pictures Entertainment (SPE)   
4                    21 (2008)  Sony Pictures Entertainment (SPE)   

index                                 Genre  Domestic_Sales  \
                                                              
0               [Action, Adventure, Sci-Fi]        94784201   
1        [Action, Adventure, Drama, Sci-Fi]       159227644   
2      [Action, Adventure, Fantasy, Sci-Fi]       127154901   
3               [Action, Adventure, Sci-Fi]       166112167   
4               [Action, Adventure, Sci-Fi]        81159365   

index  International_Sales  World_Sales Duration_min License  \
                                                               
0                175000000    269784201  1 hr 49 min   PG-13   
1                225691745    384919389  1 hr 59 min       R   
2                109195760    236350661  1 hr 47 min   PG-13   
3                625105659    791217826  2 hr 38 min   PG-13   
4                 78649005    159808370   2 hr 3 min   PG-13   

index       Release_Date  Genre_Action  ...  Genre_Horror  Genre_Music  \
                                        ...                              
0          March 5, 2008             1  ...             0            0   
1      December 25, 2019             1  ...             0            0   
2           June 5, 2003             1  ...             0            0   
3      November 11, 2009             1  ...             0            0   
4         March 28, 2008             0  ...             0            0   

index  Genre_Musical  Genre_Mystery  Genre_Romance  Genre_Sci-Fi  Genre_Sport  \
                                                                                
0                  0              0              0             0            0   
1                  0              0              0             0            0   
2                  0              0              0             0            0   
3                  0              0              0             1            0   
4                  0              0              0             0            0   

index  Genre_Thriller  Genre_War  Genre_Western  
                                                 
0                   0          0              0  
1                   0          1              0  
2                   1          0              0  
3                   0          0              0  
4                   1          0              0  

[5 rows x 30 columns]

In [16]:
# Seperate 'Release Date' feature to seperate columns 'Year', 'Month'

dt_frame.Release_Date = pd.to_datetime(dt_frame.Release_Date,format='%B %d, %Y')
dt_frame['Release_Year'] = pd.DatetimeIndex(dt_frame.Release_Date).year 
dt_frame['Release_Month'] = pd.DatetimeIndex(dt_frame.Release_Date).month 
#dt_frame['Release_Day'] = pd.DatetimeIndex(dt_frame.Release_Date).day
dt_frame.drop(['Release_Date'], axis=1, inplace = True)

In [17]:
# Remove release year for Title feature
import re
dt_frame.Title = dt_frame.Title.apply(lambda x: re.sub(r'\(\d+\)','',x))

In [18]:
# Remove release year for Title feature
import re
dt_frame.Title = dt_frame.Title.apply(lambda x: re.sub(r'\(\d+\)','',x))

In [19]:
# Convert 'Movie Runtime' into minute
"""
Steps followed are:
    1. Split the Duration columns into Hr and min. 
    2. Since some of the moview do not have min value, will set those to 0
    3. Convert the splited output to numeric.
    4. Convert Hr to min by multiplying Hr * 60. Then add that with min to get the duration in mins   
"""    
dt_frame.Duration_min = ((dt_frame.Duration_min.str.split(expand=True).iloc[:,[0,2]]).fillna(0).apply(pd.to_numeric)).apply(lambda x: (x[0]*60) + x[2],axis =1)
dt_frame.head(1)

index       Title   Distributor                        Genre  Domestic_Sales  \
                                                                               
0      10,000 BC   Warner Bros.  [Action, Adventure, Sci-Fi]        94784201   

index  International_Sales  World_Sales  Duration_min License  Genre_Action  \
                                                                              
0                175000000    269784201           109   PG-13             1   

index  Genre_Adventure  ...  Genre_Musical  Genre_Mystery  Genre_Romance  \
                        ...                                                
0                    1  ...              0              0              0   

index  Genre_Sci-Fi  Genre_Sport  Genre_Thriller  Genre_War  Genre_Western  \
                                                                             
0                 0            0               0          0              0   

index  Release_Year  Release_Month  
                                    
0              2008              3  

[1 rows x 31 columns]

In [20]:
# Set same names for Distributor 
    
dt_frame.sort_values("Distributor", inplace = True)
dt_frame.reset_index(drop=True, inplace=True)
dt_frame.loc[dt_frame.Distributor == 'DreamWorks Distribution', 'Distributor'] = 'DreamWorks' 
dt_frame.loc[dt_frame.Distributor == 'United Artists Releasing', 'Distributor'] = 'United Artists' 
dt_frame.loc[dt_frame.Distributor == 'Sony Pictures Classics', 'Distributor'] = 'Sony Pictures Entertainment' 

dt_frame.loc[dt_frame.Distributor == 'Sony Pictures Entertainment (SPE)', 'Distributor'] = 'Sony Pictures Entertainment'
dt_frame.loc[dt_frame.Distributor == 'Metro-Goldwyn-Mayer (MGM)', 'Distributor'] = 'Metro-Goldwyn-Mayer'
dt_frame.loc[dt_frame.Distributor == 'Walt Disney Studios Motion Pictures', 'Distributor'] = 'Walt Disney Studios'

dt_frame.Distributor.unique()

array(['Artisan Entertainment', 'Columbia Pictures', 'Dimension Films',
       'DreamWorks', 'FilmDistrict', 'Focus Features',
       'Fox Searchlight Pictures', 'IFC Films', 'Lionsgate',
       'Metro-Goldwyn-Mayer', 'Miramax', 'New Line Cinema',
       'Newmarket Films', 'Paramount Pictures', 'Relativity Media',
       'Revolution Studios', 'Roadside Attractions', 'STX Entertainment',
       'Screen Gems', 'Sony Pictures Entertainment',
       'Summit Entertainment', 'The Weinstein Company',
       'TriStar Pictures', 'Twentieth Century Fox', 'United Artists',
       'Universal Pictures', 'Walt Disney Studios', 'Warner Bros.'],
      dtype=object)

In [21]:
# Handel null value in License feature. 
# Here I am assuming that license issued for the similar 'Genre' can be applied to missing rows that has same 'Genre'
unique_genre_license_null_records = dt_frame[dt_frame.isna().any(axis=1)].Genre.unique()
null_index_rows = dt_frame[dt_frame.isna().any(axis=1)].index
filter_col = [col for col in dt_frame if col.startswith('Genre_')]
filter_col += ['License']
#groupby_df = dt_frame.dropna().groupby(by=['Genre','License'])
groupby_df = dt_frame.dropna().groupby(by=filter_col)

gb = groupby_df.groups
dict_genre_license= {}
for key in gb.keys():
    dict_genre_license[key[0]] = key[1]

dt_frame.iloc[null_index_rows,7] = dt_frame.iloc[null_index_rows,5].apply(lambda x: dict_genre_license[x] if x in dict_genre_license else np.nan)

#if null value still exists in the 'Licenes' feature drop those records
dt_frame.dropna(inplace = True)

print(str.format("Number of records left after drop: {0}", dt_frame.shape[0]))

Number of records left after drop: 745


In [22]:
#Final Output
dt_frame.sort_values(['Distributor','Release_Year'], inplace = True, ascending = False)
dt_frame.reset_index(drop=True, inplace=True)
dt_frame.head(5)

index                       Title   Distributor  \
0                  Birds of Prey   Warner Bros.   
1                          Joker   Warner Bros.   
2                 It Chapter Two   Warner Bros.   
3                        Shazam!   Warner Bros.   
4      Pokémon Detective Pikachu   Warner Bros.   

index                                   Genre  Domestic_Sales  \
0                         [Action, Adventure]        84158461   
1          [Adventure, Comedy, Crime, Family]       335451311   
2                 [Horror, Mystery, Thriller]       211593228   
3      [Action, Adventure, Fantasy, Thriller]       140371656   
4                            [Drama, Romance]       144105346   

index  International_Sales  World_Sales  Duration_min License  Genre_Action  \
0                117700000    201858461           109       R             1   
1                738968073   1074419384           122       R             0   
2                261500000    473093228           169       R             0   
3                225600000    365971656           132   PG-13             1   
4                289409488    433514834           104      PG             1   

index  Genre_Adventure  ...  Genre_Musical  Genre_Mystery  Genre_Romance  \
0                    1  ...              0              0              0   
1                    0  ...              0              0              0   
2                    0  ...              0              0              0   
3                    1  ...              0              0              0   
4                    1  ...              0              1              0   

index  Genre_Sci-Fi  Genre_Sport  Genre_Thriller  Genre_War  Genre_Western  \
0                 0            0               0          0              0   
1                 0            0               1          0              0   
2                 0            0               0          0              0   
3                 0            0               0          0              0   
4                 1            0               0          0              0   

index  Release_Year  Release_Month  
0              2020              2  
1              2019             10  
2              2019              9  
3              2019              4  
4              2019              5  

[5 rows x 31 columns]

In [23]:
dt_frame.columns

Index(['Title', 'Distributor', 'Genre', 'Domestic_Sales',
       'International_Sales', 'World_Sales', 'Duration_min', 'License',
       'Genre_Action', 'Genre_Adventure', 'Genre_Animation', 'Genre_Biography',
       'Genre_Comedy', 'Genre_Crime', 'Genre_Documentary', 'Genre_Drama',
       'Genre_Family', 'Genre_Fantasy', 'Genre_History', 'Genre_Horror',
       'Genre_Music', 'Genre_Musical', 'Genre_Mystery', 'Genre_Romance',
       'Genre_Sci-Fi', 'Genre_Sport', 'Genre_Thriller', 'Genre_War',
       'Genre_Western', 'Release_Year', 'Release_Month'],
      dtype='object', name='index')

In [24]:
dt_frame.Release_Year.min()

1977

In [25]:
dt_frame.Release_Year.max()

2021

Summary:

1. In original data there where 173 records that had null value in 'License' feature. We had to dropped 55 records since we where not able to find any past moview licenses with same genre. Now we have 863 unique records
2. Add new columns that might help in out analysis. Like now we have seperate column for each Gener 
3. We have movies records from year 1953 to 2021